pep-0699.rst

```diff
@@ -64,10 +64,11 @@ Certain extensions use ``ma_version_tag`` for fast dictionary or globals
 lookups. For example,
 `Cython uses the field for fast dynamic module variable lookups <https://github.com/cython/cython/blob/169876872f3cb6198971a1db07e5b8a9d12b3dac/Cython/Utility/ObjectHandling.c#L1556>`_.
 
-Since this field is private and makes no guarantees on backwards compatibility,
-this PEP proposes to repeal :pep:`509` immediately within the next CPython
-version. A function for fast cached dictionary lookups may be provided as an
-alternative feature to replace directly accessing ``ma_version_tag``.
+This PEP proposes to emit a compiler warning when accessing ``ma_version_tag``.
+After two consecutive version releases with warnings, :pep:`509` will be fully
+repealed, in line with :pep:`387`. A function for fast cached dictionary
```

Comment:
```suggestion
After two consecutive version releases with warnings, ``ma_version_tag``
will be change or removed, in line with :pep:`387`.
A function for fast cached dictionary
```

IMO, it is clearer to state this in terms of the effect on the implementation as opposed to the procedural motions that are of lesser importance to users, though perhaps the one challenge is you aren't entirely sure what these will be—if you're just going to introduce a new field for this rather than re-using `ma_version_tag` as initially planned, you could revise the line to instead say

```suggestion
After two consecutive version releases with warnings, ``ma_version_tag``
will be removed, in line with :pep:`387`. A function for fast cached dictionary
```

Reply:
However, we don't know if we're removing ``ma_version_tag``. The whole point of the PEP is just that the specification is undone, not that we're removing the field.

Reply:
Right—thought saying "the PEP is fully repealed" still isn't necessarily clear or obviously pertinent when it comes to a specific Python implementation version. I've tweaked the first option to read `_may_ be changed or removed`, to make it clear what the practical effect on CPython implementation users of "the PEP being repealed", that something _may_ change or be removed after that point, rather than committing to any particular course of action.